### PR TITLE
[READWISE] RxDB patch

### PR DESCRIPTION
**Pull request by eliias**
## Todos 
- [ ] Tests
- [ ] Documentation
- [ ] Typings
- [ ] Changelog

### CODE REVIEW

File: dist/types/rx-query.d.ts [hunk 1]
```diff
@@ -27,6 +32,12 @@ export declare class RxQueryBase<RxDocType, RxQueryResult, OrmMethods = {}, Reac
     _latestChangeEvent: -1 | number;
     _lastExecStart: number;
     _lastExecEnd: number;
+    _limitBufferSize: number | null;
+    _limitBufferResults: RxDocumentData<RxDocType>[] | null;
+    _persistentQueryCacheResult?: string[] | string;
```
Comment (TristanH) [inline dist/types/rx-query.d.ts:37]: i hope this stuff wasn't a hassle to carry over lol. given we don't use it rn

File: src/event-reduce.ts [hunk 2]
```diff
@@ -112,6 +125,48 @@ export function getQueryParams<RxDocType>(
     );
 }
 
+// This catches a specific case where we have a limit query (of say LIMIT items), and then
```
Comment (TristanH) [inline src/event-reduce.ts:128]: ooc have you tested to confirm the limit buffer stuff works? i guess we have tests for it

**Comment by TristanH** [comment]
should this PR be into the `readwise ` branch instead of `master`?

File: src/event-reduce.ts (hunk 2 above)
Comment (eliias) [reply src/event-reduce.ts:128]: I did not test anything tbh. I just rebased everything we h ad into a commit and resolved the merge conflicts. I think this comment is from last year :-).

**Comment by eliias** [comment]
> should this PR be into the `readwise ` branch instead of `master`?

@TristanH yeah I was wondering how you wanna deal with this rxdb15 upgrade. should we do readwise15 or we comfy with just reusing readwise?

**Comment by TristanH** [comment]
@eliias im fine with just updating `readwise`. we reference a commit hash in our package.json anyways: https://github.com/readwiseio/rxdb_fork/commit/53f893bfc7d847d258cf3ff16bce80c09ad38455

i guess the problem with reviewing the PR that was is that all the rxdb code updates would show in that PR?

**Comment by eliias** [comment]
> @eliias im fine with just updating `readwise`. we reference a commit hash in our package.json anyways: [53f893b](https://github.com/readwiseio/rxdb_fork/commit/53f893bfc7d847d258cf3ff16bce80c09ad38455)
> 
> i guess the problem with reviewing the PR that was is that all the rxdb code updates would show in that PR?

Yep, this PR includes all our changes against master, it was impossible to rebase all our commits against 1000+ rxdb commits (impossible = infinitely more work).

File: src/event-reduce.ts (hunk 2 above)
Comment (TristanH) [reply src/event-reduce.ts:128]: ya it is. just seeing the limit buffer code reminded me that we need to make sure that still works. it's all my custom code to make rxdb faster

Comment (eliias) [reply src/event-reduce.ts:128]: Yeah I can test more, there was nothing obvious that broke so far though and the code is/was pretty similar still in rx-query (and therefore I somewhat assume event-reduce to behave similar).

**Comment by TristanH** [comment]
hmm ok, well i guess i worry about merging into `readwise` then as we could lose the old commit that master points at right now?

we could rename the old branch to `readwise_2023`?

**Comment by TristanH** [comment]
back to the original (separate) question, i wish i had a way to review the actual changes, but i see why that's probably not doable here... will just cross reference with our old code

**Comment by TristanH** [comment]
ok @eliias reviewed it mostly, is there anywhere (i assume rx-query.ts) i should really look relevant to your changes?

**Comment by eliias** [comment]
> ok @eliias reviewed it mostly, is there anywhere (i assume rx-query.ts) i should really look relevant to your changes?

There are no real changes tbh, this was a rebase exercise. I don't know how to improve the review experience beyond what we have here, but I was hoping you remember your changes and I remember mine, and if we agree it does the same as before we good to ship.

**Pull request closed without merging**